src/Symfony/Component/Process/Process.php

```diff
@@ -1237,14 +1237,31 @@ public static function isPtySupported()
             return $result;
         }
 
-        if ('\\' === DIRECTORY_SEPARATOR) {
+        if (self::isWindowsPlatform()) {
             return $result = false;
         }
 
         return $result = (bool) @proc_open('echo 1', array(array('pty'), array('pty'), array('pty')), $pipes);
     }
 
     /**
+     * Returns whether the current process is run on Windows platform (Command Prompt, PowerShell) or not.
+     * 
+     * Keep in mind, that Windows users can use Unix shells (like the git-shell) with TTY/PTY support,
+     * in this case, this function will return 'false'.
+     *
+     * @return bool
+     */
+    public static function isWindowsPlatform()
```

Comment:
this should not be a public method.

and there are several issues related to this method: you used it in all places wanting to detect Windows, but this method is not about detecting Windows (there is a mismatch between the name and the implementation, which caused you to introduce bugs by expecting the behavior of its name in many places).